### ISSUE TITLE
TRY300: Recommend putting code after the try/except instead of inside a else

### ISSUE THREAD

**Issue by cbornet** [astral-sh/ruff#13885]
When an exception handler re-raises or returns, it seems to me that it's better to put the "else" code as a follow-up to the try/except than in a "else".

Eg for the code
```python
import logging


def reciprocal(n):
    try:
        rec = 1 / n
        print(f"reciprocal of {n} is {rec}")
        return rec
    except ZeroDivisionError:
        logging.exception("Exception occurred")
        raise
```
you get `TRY300 Consider moving this statement to an `else` block`
but the best for me would be to write:
```python
import logging


def reciprocal(n):
    try:
        rec = 1 / n
    except ZeroDivisionError:
        logging.exception("Exception occurred")
        raise
    print(f"reciprocal of {n} is {rec}")
    return rec
```

BTW, the current example also seems incorrect as `reciprocal` returns an implicit `None` (See [RET503](https://docs.astral.sh/ruff/rules/implicit-return/) even though it doesn't detect it)
